src/fireedge/src/server/utils/jwt.js

```diff
@@ -15,39 +15,62 @@
  * ------------------------------------------------------------------------- */
 
 const jwt = require('jwt-simple')
-const { messageTerminal } = require('./general')
 const speakeasy = require('speakeasy')
+const { messageTerminal } = require('server/utils/general')
+const { JWTError, MissingFireEdgeKeyError } = require('server/utils/errors')
 
 /**
  * Create a JWT.
  *
- * @param {object} param0 - object of data to JWT {ID, USER, Opennebula_token}
- * @param {string} param0.id - user ID
- * @param {string} param0.user - username
- * @param {string} param0.token - token opennebula
- * @param {number} iat - epoch create time (now)
- * @param {number} exp - epoch expire time
+ * @param {object} jwtData - object of data to JWT {ID, USER, Opennebula_token}
+ * @param {string} jwtData.id - user ID
+ * @param {string} jwtData.user - username
+ * @param {string} jwtData.token - token opennebula
  * @returns {string} JWT
  */
-const createJWT = (
-  { id: iss, user: aud, token: jti },
-  iat = '',
-  exp = ''
-) => {
+const createJWT = ({ id, user, token }) => {
+  if (id && user && token) {
+    return jwtEncode({
+      iss: id, // user ID
+      aud: user, // user name
+      jti: token, // token
+    })
+  }
+}
+
+/**
+ * Encode JWT.
+ *
+ * @param {object} payload - data object
+ * @returns {object} - jwt or null
+ */
+const jwtEncode = (payload = {}) => {
   let rtn = null
-  if (iss && aud && jti && iat && exp && global && global.paths && global.paths.FIREEDGE_KEY) {
-    const payload = {
-      iss,
-      aud,
-      jti,
-      iat,
-      exp
-    }
+  if (global && global.paths && global.paths.FIREEDGE_KEY) {
     rtn = jwt.encode(payload, global.paths.FIREEDGE_KEY)
   }
+
   return rtn
 }
 
+/**
+ * Decode JWT.
+ *
+ * @param {string} token - token JWT
+ * @returns {object} data JWT
+ */
+const jwtDecode = (token = '') => {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Avoid using `==` and `===` operators with sensitive data fields like passwords, api keys, tokens etc. as they are vulnerable to timing attack. Consider using `crypto.timingSafeEqual` or `secure-compare` for time-safe string comparison.

[Learn more](https://cwe.mitre.org/data/definitions/208.html)